### PR TITLE
Use pro server and skynet-api-key in nodejs CI

### PR DESCRIPTION
**Pull request by m-cat**
# PULL REQUEST

## Overview

Now that nodejs-skynet makes more real network requests in its test suite, we
have started encountering a lot more failed tests in CI. This can be largely
mitigated by using the pro server with an API key.

### NOTE

This PR ended up being a lot bigger than I expected it to be. There were a number of issues preventing the API key from working correctly - so this PR has a large number of fixes due to my debugging. There is even a refactor of default options, which was necessary to prevent a circular dependency situation (`skydb_v2.js` pulls in `DEFAULT_UPLOAD_OPTIONS` which was hitting `upload.js` before `SkynetClient` was defined. The refactor fixed this by moving `DEFAULT_UPLOAD_OPTIONS` to `defaults.js`).

Please let me know if anything's not clear or if anything can be improved.

## Checklist

- [x] All git commits are signed. (REQUIRED)
- [x] All new methods or updated methods have clear docstrings.
- [x] Testing added or updated for new methods.
- [x] Verified if any changes impact the WebPortal Health Checks.
- [x] Appropriate documentation updated.
- [x] Changelog file created.

## Issues Closed

Closes SKY-904
Closes SKY-910

### CODE REVIEW

**Comment by m-cat** [comment]
This PR ended up being a lot bigger than I expected it to be. There were a number of issues preventing the API key from working correctly - so this PR has a large number of fixes due to my debugging. There is even a refactor of default options, which was necessary to prevent a circular dependency situation (`skydb_v2.js` pulls in `DEFAULT_UPLOAD_OPTIONS` which was hitting `upload.js` before `SkynetClient` was defined. The refactor fixed this by moving `DEFAULT_UPLOAD_OPTIONS` to `defaults.js`).

Please let me know if anything's not clear or if anything can be improved.

File: integration/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const axios = require("axios");
+const axiosRetry = require("axios-retry");
+const { SkynetClient, DEFAULT_SKYNET_PORTAL_URL } = require("../index");
+
+// retry if we're getting rate limited
```
Comment (m-cat) [inline integration/index.js:5]: Copied from skynet-js. Figured, why not?

File: integration/skydb.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const { client, portal } = require(".");
```
Comment (m-cat) [inline integration/skydb.test.js:1]: Moved into new `integration` folder and removed unnecessary `describe` blocks. I thought those were causing tests to run in parallel but it didn't make a difference - it is, however a bit cleaner without them. There were also some `it` tests with duplicate names which I renamed.

File: integration/index.js (hunk 1 above)
Comment (parajbs) [reply integration/index.js:5]: hello m-cat,

if it works.
I also used the test from "skynet-js" where the
"axios-mock-adapter" is used.

File: integration/skydb.test.js (hunk 2 above)
Comment (parajbs) [reply integration/skydb.test.js:1]: hello m-cat,

I myself have done a test with several portals and then, if it was feasible, used different portals, so the tests ran quickly.
But I'm not sure if you mean these.

**Comment by m-cat** [comment]
> Sorry for the errors caused by my programming style.

Hey @parajbs, don't worry about it! The SkyDB PR was really big and I missed these bugs, too. And the way that options are passed along is really not obvious - it was hard to catch unless you actually tried running on a different portal with an API key.

> Why don't we put all tests in the integration folder, because my new RP with new functions, many new tests are included, and it is then tidier and looks better.

Yeah, any tests that make network requests should go in there. Unit tests or tests with mocks should stay where they are.

> I think the code looks good, but I can't say whether it works with the pro server and skynet-api-key. You have to test these.

It's tested by the CI. It works well, though I had to increase the test timeout from 5 seconds to 2 minutes - @kwypchlo @MSevey is it a known issue that the registry is really slow on the pro portal?

>  The problem also occurs with accounts from "siasky.net" and "skynetFree.net" too, or only with pro server?

Not sure what you mean @parajbs. You don't need an API key for siasky.net, but the pro server should be the most stable one which is why I chose it for the CI tests. Hopefully this cuts down significantly on the amount of test failures we're seeing.

**Comment by parajbs** [comment]
hello m-cat,

At the beginning of skynet I had registered an account with "siasky.net" before skynetfree existed. But today I can no longer log in, but this can also be due to a wrong password.
That's why I asked if it was also at "siasky.net", but I didn't know that the accounts were no longer there. Was just a test account for me to see how it works. That's why I haven't used it for a long time.

And what about skynetFree?
There at skynetfree I also saw something with api-key and sonsor-keys on my brother's account.

If skynetFree also had the problem, then yes I can
 maybe test it.

**Comment by MSevey** [comment]
> It's tested by the CI. It works well, though I had to increase the test timeout from 5 seconds to 2 minutes - @kwypchlo @MSevey is it a known issue that the registry is really slow on the pro portal?

@m-cat eu-pol-102 has timeout issues right now. the other servers in the pro portal should be fine though.

**Review by MSevey — APPROVED** [review]

**Review by kwypchlo — APPROVED** [review]
